### ISSUE TITLE
Configure `pipe` for child thread

### ISSUE THREAD

**Issue by moteus** [zeromq/lzmq#37]
We need way to configure protocol(tcp/ipc/inpoc) name(IP/path/name) and type(PAIR/DEALER/REQ)
To configure pipe type we some time needs provide type for parent and child socket.
Also there may be use case when you run worker threads and whant to connect their pipes to some manager. In this case we should tell to `fork` that it should not create parent pipe and just crate child pipe and connect it to existing endpoint
```Lua
local ep = "inproc://some.manager"
manager = Manager.new(ep)
for i = 1, 10 do
  zthreads.xfork{
    -- worker thread manageg by `manager` object
    function(pipe) pipe:send"READY" ... end;
    pipe = { ? = ep}
  }
end
```

I suggest use `pipe` option instead of `endpoint_transport` and `endpoint_name`
At first it can be protocol(transport) or full endpoint address.
e.g. `pipe="tcp"` `pipe="ipc://some/pipe/name"`
Extract transport part from string is easy task. And provide name without protocol is quite useless thing.
In the next version we could extend `pipe` argument to accept table
```lua
pipe = {
  endpoint = -- same as `pipe` string argument
  ... other options to configure parent/child pipe
}
```

/cc @johnfoconnor

**Comment by johnfoconnor** [comment]
this seems reasonable. to clarify we are saying

```
{
pipe = "some string" -- transport or full endpoint
}
```

what about for specifying the type? we need to specify the type & connect/bind for both the parent and child. is using the same form as http://moteus.github.io/lzmq/modules/lzmq.html#context:socket sufficient?

**Comment by moteus** [comment]
Yes. For now `pipe` is transport or full endpoint e.g.`pipe='tcp'` or `pipe='tcp://127.0.0.1:12345'`

>what about for specifying the type?

I do not think about that yet. But in this case `pipe` will be a table with options.
Using just ctor table is possible, but you need also think about dynamic endpint.
Also may be you will need use existing socket. I do not want implement this without any real usecases.

For now I think about implement interface similar to zthreads but for processes.

**Comment by moteus** [comment]
Closed by #38
